src/Data/MIME/EncodedWord.hs

```diff
@@ -144,3 +146,17 @@ decodeEncodedWords charsets s =
       (view (charsetDecoded charsets) w)
 
     merge = foldMap (either decodeLenient id)
+
+-- | This function encodes necessary parts of some text
+--
+-- Currently turns the whole text into a single encoded word, if necessary.
+encodeEncodedWords :: T.Text -> B.ByteString
+encodeEncodedWords s
+  | cteName == "7bit" = s'
+  | cteName == "base64" = "=?utf-8?B?" <> s' <> "?="
+  | cteName == "quoted-printable" = "=?utf-8?Q?" <> s' <> "?="
+  | otherwise = "=?utf-8?X?" <> s' <> "?="
+  where
+    s' = C.filter (`notElem` ['\r', '\n']) $ review (clonePrism cte) bytes
+    (cteName, cte) = chooseTransferEncoding bytes
```

Comment:
Thanks for the PR.  The 'Q' encoding is not quite the same as Quoted-Printable.  So we ought to have a dedicated function `chooseEncodedWordEncoding :: T.Text -> Maybe (TransferEncodingName, TransferEncoding)` that provides the `B` or `Q` encoding prism if encoding is needed.  At the call site, for the `Just` case, construct an `EncodedWord` and apply `serialiseEncodedWord` to it.

The tests look good.

Reply:
I'm happy to leave these changes to you, or I can take it from here if you wish.

Reply:
I've made these changes now, I think.

Reply:
@singpolyma thanks, I'll review it in the next day or so.